stubs/gdb/gdb/__init__.pyi

```diff
@@ -0,0 +1,695 @@
+
+class Field:
+
+    bitpos: int
+    enumval: int
+    name: str | None
+    artificial: bool
+    is_base_class: bool
+    bitsize: int
+    type: Type
+    parent_type: Type
+
+TYPE_CODE_PTR: int
+TYPE_CODE_ARRAY: int
+TYPE_CODE_STRUCT: int
+TYPE_CODE_UNION: int
+TYPE_CODE_ENUM: int
+TYPE_CODE_FLAGS: int
+TYPE_CODE_FUNC: int
+TYPE_CODE_INT: int
+TYPE_CODE_FLT: int
+TYPE_CODE_VOID: int
+TYPE_CODE_SET: int
+TYPE_CODE_RANGE: int
+TYPE_CODE_STRING: int
+TYPE_CODE_BITSTRING: int
+TYPE_CODE_ERROR: int
+TYPE_CODE_METHOD: int
+TYPE_CODE_METHODPTR: int
+TYPE_CODE_MEMBERPTR: int
+TYPE_CODE_REF: int
+TYPE_CODE_RVALUE_REF: int
+TYPE_CODE_CHAR: int
+TYPE_CODE_BOOL: int
+TYPE_CODE_COMPLEX: int
+TYPE_CODE_TYPEDEF: int
+TYPE_CODE_NAMESPACE: int
+TYPE_CODE_DECFLOAT: int
+TYPE_CODE_INTERNAL_FUNCTION: int
+
+# Pretty Printing
+
+class _PrettyPrinter(Protocol):
+    # TODO: The "children" and "display_hint" methods are optional for
+    # pretty-printers. Unfortunately, there is no such thing as an optional
+    # method in Mypy at the moment.
+    #
+    # def children(self) -> Iterator[tuple[str, _ValueOrNative]]: ...
+    # def display_hint(self) -> str | None: ...
+    def to_string(self) -> str | LazyString: ...
+
+_PrettyPrinterLookupFunction: TypeAlias = Callable[[Value], Optional[_PrettyPrinter]]
+
+def default_visualizer(value: Value) -> _PrettyPrinter | None: ...
+
+# Selecting Pretty-Printers
+
+pretty_printers: list[_PrettyPrinterLookupFunction]
+
+# Filtering Frames
+
+class FrameFilter:
+
+    name: str
+    enabled: bool
+    priority: int
+
+    def filter(self, iterator: Iterator[_FrameDecorator]) -> Iterator[_FrameDecorator]: ...
+
+# Decorating Frames
+
+class _SymValueWrapper(Protocol):
+    def symbol(self) -> Symbol | str: ...
+    def value(self) -> _ValueOrNative | None: ...
+
+class _FrameDecorator(Protocol):
+    def elided(self) -> Iterator[Frame] | None: ...
+    def function(self) -> str | None: ...
+    def address(self) -> int | None: ...
+    def filename(self) -> str | None: ...
+    def line(self) -> int | None: ...
+    def frame_args(self) -> Iterator[_SymValueWrapper] | None: ...
+    def frame_locals(self) -> Iterator[_SymValueWrapper] | None: ...
+    def inferior_frame(self) -> Frame: ...
+
+# Unwinding Frames
+
+class PendingFrame:
+    def read_register(self, reg: str | RegisterDescriptor | int) -> Value: ...
+    def create_unwind_info(self, frame_id: object) -> UnwindInfo: ...
+    def architecture(self) -> Architecture: ...
+    def level(self) -> int: ...
+
+class UnwindInfo:
+    def add_saved_register(self, reg: str | RegisterDescriptor | int, value: Value) -> None: ...
+
+class Unwinder:
+    name: str
+    enabled: bool
+
+    def __call__(self, frame: Frame) -> UnwindInfo | None: ...
+
+# Xmethods: the API is defined in the "xmethod" module
+
+# Inferiors
+
+def inferiors() -> tuple[Inferior, ...]: ...
+def selected_inferior() -> Inferior: ...
+
+BufferType: TypeAlias = Union[memoryview, bytes]
```

Comment:
```suggestion
BufferType: TypeAlias = memoryview | bytes
```

Reply:
Probably should be `_typeshed.ReadableBuffer` (I haven't checked the code though)

Reply:
> Probably should be `_typeshed.ReadableBuffer` (I haven't checked the code though)

I also haven't checked the code, but yeah, sounds plausible :)

Reply:
I’ve double-checked passing a `mmap.mmap` instance: it looks like it works, thanks!